gcp-auth-extension/src/main/java/io/opentelemetry/contrib/gcp/auth/GcpAuthCustomizerProvider.java

```diff
@@ -0,0 +1,169 @@
+    customizeTracer(model, headerMap, configProperties);
+  }
+
+  static ConfigProperties getConfigProperties(OpenTelemetryConfigurationModel model) {
+    return new DeclarativeConfigPropertiesBridgeBuilder()
+        .resolveInstrumentationConfig(SdkConfigProvider.create(model).getInstrumentationConfig());
+  }
+
+  private static void customizeMeter(
+      OpenTelemetryConfigurationModel model,
+      Map<String, String> headerMap,
+      ConfigProperties configProperties) {
+    MeterProviderModel meterProvider = model.getMeterProvider();
+    if (meterProvider == null) {
+      return;
+    }
+
+    if (shouldConfigureExporter(
+        SIGNAL_TYPE_METRICS, SIGNAL_TARGET_WARNING_YAML_FIX_SUGGESTION, configProperties)) {
+      for (MetricReaderModel reader : meterProvider.getReaders()) {
+        if (reader.getPeriodic() != null) {
+          addAuth(meterModelHeaders(reader.getPeriodic().getExporter()), headerMap);
+        }
+      }
+    }
+  }
+
+  private static List<List<NameStringValuePairModel>> meterModelHeaders(
+      @Nullable PushMetricExporterModel exporter) {
+    ArrayList<List<NameStringValuePairModel>> list = new ArrayList<>();
+    if (exporter == null) {
+      return list;
+    }
+    OtlpGrpcMetricExporterModel grpc = exporter.getOtlpGrpc();
+    if (grpc != null) {
+      list.add(grpc.getHeaders());
+    }
+    OtlpHttpMetricExporterModel http = exporter.getOtlpHttp();
+    if (http != null) {
+      list.add(http.getHeaders());
+    }
+    return list;
+  }
+
+  private static void customizeTracer(
+      OpenTelemetryConfigurationModel model,
+      Map<String, String> headerMap,
+      ConfigProperties configProperties) {
+    TracerProviderModel tracerProvider = model.getTracerProvider();
+    if (tracerProvider == null) {
+      return;
+    }
+
+    if (shouldConfigureExporter(
+        SIGNAL_TYPE_TRACES, SIGNAL_TARGET_WARNING_YAML_FIX_SUGGESTION, configProperties)) {
+      for (SpanProcessorModel processor : tracerProvider.getProcessors()) {
+        BatchSpanProcessorModel batch = processor.getBatch();
+        if (batch != null) {
+          addAuth(spanExporterModelHeaders(batch.getExporter()), headerMap);
+        }
+        SimpleSpanProcessorModel simple = processor.getSimple();
+        if (simple != null) {
+          addAuth(spanExporterModelHeaders(simple.getExporter()), headerMap);
+        }
+      }
+    }
+  }
+
+  private static List<List<NameStringValuePairModel>> spanExporterModelHeaders(
+      @Nullable SpanExporterModel exporter) {
+    ArrayList<List<NameStringValuePairModel>> list = new ArrayList<>();
+
+    if (exporter == null) {
+      return list;
+    }
+    OtlpGrpcExporterModel grpc = exporter.getOtlpGrpc();
+    if (grpc != null) {
+      list.add(grpc.getHeaders());
+    }
+    OtlpHttpExporterModel http = exporter.getOtlpHttp();
+    if (http != null) {
+      list.add(http.getHeaders());
+    }
+    return list;
+  }
+
+  private static void addAuth(
```

Comment:
Question: The `addAuth` method seems to add a fixed key-value pair to a list of headers associated with the span exporter / metric exporter. 

In this case, I don't see how a token refresh for the auth headers work or am I missing something here ?

Reply:
great catch - I missed the refresh part completely - I have to check how this is even supposed to work